### ISSUE TITLE
🐛 `Database` check reports false failures on stale connections

### ISSUE THREAD

**Issue by czue** [KristianOellegaard/django-health-check#755]
### Bug Description

I started getting an overwhelming number of false positive failures on my database connction of the form:

```
Unavailable: the connection is closed
```

I / Claude chased this down to it being a stale connection *handle*, which is something that Django can self-recover from.

Here's more details (written by Claude):

## Why it happens

1. **`checks.Database.run` is synchronous**, so `HealthCheck.get_result` dispatches it via
   `loop.run_in_executor(executor, self.run)` (`base.py`), and `HealthCheckView.get_executor`
   returns `contextlib.nullcontext(None)` — so `executor is None` and the work lands on the
   **event loop's default thread pool**. Those threads are long-lived and sit entirely outside
   the request/response cycle.

2. **Django's `connections` is `Local(thread_critical=True)`**, which in a thread with no running
   event loop is a plain `threading.local()` (asgiref `local.py`). So each pool thread keeps its
   own `DatabaseWrapper`. Django's `close_old_connections` is wired to the `request_started` /
   `request_finished` signals, which never fire on those threads — so nothing ever cleans up or
   revalidates a connection parked there.

3. **`Database.run` uses `connection.temporary_connection()`**, which is
   `must_close = self.connection is None`. It closes only what *it* opened. Any connection already
   present on that thread is reused **with no liveness check** — `_cursor()` only pings via
   `close_if_health_check_failed()` when `CONN_HEALTH_CHECKS` is enabled *and* `health_check_done`
   is `False`, and `health_check_done` is reset only by `connect()` and
   `close_if_unusable_or_obsolete()`, neither of which runs on those pool threads.

Net effect: if a connection is present on a default-executor thread and has gone stale, the check
reports a hard failure with no attempt to recover, in a place where Django's usual recovery
machinery is structurally unable to reach.

### Expected Behavior

The check should retrieve/recover from stale connection handles the same way a normal Django application would.

### Version

4.4.1

### Steps to Reproduce

(This section was written by Claude).

Against a real Postgres. Do **not** use sqlite — `sqlite3/base.py::is_usable()` unconditionally
returns `True`, which masks the `CONN_HEALTH_CHECKS` path and makes the bug look different than
it is.

```python
import django
from django.conf import settings

settings.configure(
    DEBUG=False,
    DATABASES={"default": {
        "ENGINE": "django.db.backends.postgresql",
        "NAME": "postgres", "USER": "postgres", "PASSWORD": "postgres",
        "HOST": "localhost", "PORT": "5432",
        "CONN_MAX_AGE": 0,
    }},
    INSTALLED_APPS=["django.contrib.contenttypes"],
    USE_TZ=True,
)
django.setup()

from django.db import connections
from health_check import checks

# Simulate what a server-side termination, a pooler timeout, or a resumed VM
# leaves behind: Django still holds a DatabaseWrapper whose socket is dead.
conn = connections["default"]
conn.connect()
conn.connection.close()          # driver-level close, behind Django's back
assert conn.connection is not None

checks.Database().run()
```

**Expected:** the check notices the handle is unusable, reconnects, and reports OK — the database
is fine and any ordinary Django view would have served the request.

**Actual:**

```
health_check.exceptions.ServiceUnavailable: Unavailable: the connection is closed
```

**Comment by codingjoe** [comment]
https://github.com/codingjoe/django-health-check/pull/756#issuecomment-5178886603

**Comment by codingjoe** [comment]
Small addition:

I would suggest looking into why your event loop's thread pool has unclosed connections. Django itself usually closes its threaded connections; this might be an issue in a 3rd party package or your 1st party code.
